packages/ui/src/views/w3m-wallet-explorer-view/index.ts

```diff
@@ -13,26 +15,64 @@ import '../../components/w3m-modal-header'
 import '../../components/w3m-spinner'
 import '../../components/w3m-text'
 import '../../components/w3m-wallet-button'
+import { SEARCH_ICON } from '../../utils/Svgs'
 import { global } from '../../utils/Theme'
-import { getErrorMessage, getShadowRootElement, preloadImage } from '../../utils/UiHelpers'
+import {
+  debounce,
+  getErrorMessage,
+  getShadowRootElement,
+  preloadImage
+} from '../../utils/UiHelpers'
 import styles, { dynamicStyles } from './styles'
 
+const PAGE_ENTRIES = 40
+
 @customElement('w3m-wallet-explorer-view')
 export class W3mWalletExplorerView extends LitElement {
   public static styles = [global, styles]
 
   // -- state & properties ------------------------------------------- //
-  @state() private loading = !ExplorerCtrl.state.wallets.listings.length
-  @state() private firstFetch = !ExplorerCtrl.state.wallets.listings.length
+  @state() private firstFetch = true
+  @state() private search = ''
   @state() private endReached = false
+  @state() private listingResponse: ListingResponse = { listings: [], total: 0 }
```

Comment:
Why are these stored here as opposed to `ExplorerCtrl`? If you need them separate, let's just create new entry there like `searchResult`?

Reply:
Since we are calling `getPaginatedWallets` from 2 different places, I've opted to do this so we control when the listings are loaded into UI (eg, after preloading the images, etc)